### PR TITLE
AMD ROCm - HIP update and bump up version to 3.9.0 for rccl,debug agent, hip-rocclr and atmi

### PR DESCRIPTION
**Pull request by arjun-raj-kuppala**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hip/package.py [hunk 1]
```diff
@@ -55,8 +56,11 @@ def setup_run_environment(self, env):
         env.set('HIP_CLANG_PATH', self.spec['llvm-amdgpu'].prefix.bin)
         env.set('HSA_PATH', self.spec['hsa-rocr-dev'].prefix)
         env.set('ROCMINFO_PATH', self.spec['rocminfo'].prefix)
+        env.set('HIP_PATH', self.prefix)
         env.set('DEVICE_LIB_PATH',
-                self.spec['rocm-device-libs'].prefix.lib)
+                self.spec['rocm-device-libs'].prefix.amdgcn.bitcode)
+        env.set('HIPCC_COMPILE_FLAGS_APPEND',
+                '--rocm-path={}'.format(self.prefix))
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/hip/package.py]: --rocm-path={}  -change to --rocm-path={0}.format

**Review by srekolam — CHANGES_REQUESTED** [review]
i have added a comment for hip package.py,please address it.

**Review by srekolam — APPROVED** [review]

File: var/spack/repos/builtin/packages/hip/package.py (hunk 1 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hip/package.py]:
```suggestion
                '--rocm-path={0}'.format(self.prefix))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by arjun-raj-kuppala** [comment]
@adamjstewart Can you help merge this ?

**Pull request merged**